mdr/mdr.py

```diff
@@ -73,7 +92,21 @@ def transform(self, features):
             Constructed features from the provided feature matrix
 
         """
-        return 0
+        new_feature = np.zeros(shape=(features.shape[0],1))
+        for i in range(features.shape[0]):
+            feature_instance = tuple(map(tuple, features[i]))
+            if feature_instance in feature_map:
+                counts = feature_map[feature_instance]
+                if counts[0] > counts[1]:
```

Comment:
This comparison needs to be made based on ratios, not counts. You calculate the overall ratio of cases to controls during `fit`, and store that ratio in an internal variable (`self.class_ratio`). Then the comparison will look something like:

```
if counts[0] / counts[1] > self.class_ratio:
    # ... etc.
```